src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1042,6 +1042,28 @@ public static bool IsDefaultExpressionContext(this SyntaxTree syntaxTree, int po
             return false;
         }
 
+        public static bool IsParamsModifierContext(
+            this SyntaxTree syntaxTree,
+            int position,
+            SyntaxToken tokenOnLeftOfPosition,
+            CancellationToken cancellationToken)
+        {
+            var token = tokenOnLeftOfPosition;
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            if (syntaxTree.IsParameterModifierContext(position, tokenOnLeftOfPosition, cancellationToken))
+                return true;
+
+            if ((token.IsKind(SyntaxKind.OpenBracketToken) || token.IsKind(SyntaxKind.CommaToken)) &&
+                token.Parent.IsKind(SyntaxKind.BracketedParameterList) &&
+                token.Parent.IsParentKind(SyntaxKind.IndexerDeclaration))
```

Comment:
please don't mix &&'s and ||'s in a single condition.  It gets difficult to read.  This can be written as:

```c#
if (token.IsKind(SyntaxKind.OpenBracketToken) || token.IsKind(SyntaxKind.CommaToken))
{
    return token.IsParentKind(SyntaxKind.BracketedParameterList);
}

return false;
```